core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -257,8 +259,12 @@ object Battle {
                 if (defeatedUnitYieldSourceType == "Cost") unitCost else unitStr
             val yieldAmount = (yieldTypeSourceAmount * yieldPercent).toInt()
 
-            val stat = Stat.valueOf(unique.params[3])
-            civUnit.getCivInfo().addStat(stat, yieldAmount)
+            val resourceName = unique.params[3]
+            val resource = Stat.safeValueOf(resourceName) ?:
+            SubStat.safeValueOf(resourceName) ?:
+            civUnit.getCivInfo().gameInfo.ruleset.tileResources[resourceName]
+            if (resource == null) continue
```

Comment:
This is repeated at least 3 times, sounds like it should be a function getResource(resourceName)